lib/devise/models/validatable.rb

```diff
@@ -62,8 +76,53 @@ def email_required?
         true
       end
 
+      # Make these instance methods so the default Devise.password_requires_<> 
+      #can be overridden
+      def password_complexity
+        self.class.password_complexity
+      end
+
+      def password_requires_lowercase
+        password_complexity[:require_lower]
+      end
+
+      def password_requires_uppercase
+        password_complexity[:require_upper]
+      end
+      
+      def password_requires_digit
+        password_complexity[:require_digit]
+      end
+
+      def password_requires_special_character
+        password_complexity[:require_special_character]
+      end
+
+      def allowed_special_characters
+        password_complexity[:allowed_special_characters]
+      end
+
+      def password_contains_special_character    
```

Comment:
@salzig @fthobe this allows both approaches: `\W` and check based on configured special chars 👍